csharp/src/Apache.Arrow/ChunkedArray.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Apache.Arrow;
+using Apache.Arrow.Types;
+
+namespace Apache.Arrow
+{
+    /// <summary>
+    /// A data structure to manage a list of primitive Array arrays logically as one large array
+    /// </summary>
+    internal class ChunkedArray
+    {
+        public IList<Array> Arrays { get; }
+        public readonly IArrowType DataType;
+        public readonly long Length;
+        public readonly long NullCount;
```

Comment:
All 3 of these should be properties. In general, don't expose fields externally to a class.